### PR TITLE
Bug 1158520 - Fix remaining Gallery jshint issues

### PR DESCRIPTION
**Pull request by KevinGrandon**
https://bugzilla.mozilla.org/show_bug.cgi?id=1158520

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=373f28ab044880e417cfd3a068b7658af7579abb

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=dbd6751f8b3d6ff4c141caf5edcf84a7fb8e9cb7

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e7e364523ea5ed2d6d7bde9d1ac71f468856b96a

File: apps/gallery/js/ImageEditor.js [hunk 1]
```diff
@@ -28,6 +51,37 @@ $('edit-save-button').onclick = saveEditedImage;
...
 editAspectMoreButtons.forEach(function(b) {b.onclick = editAspectMoreHandler;});
 
+/*
+ * These arrays represent matricies for image processing effects. They
+ * are passed to the ImageProcessor.setMatrix() method to specify color
+ * transformations of an image. The property names in this object are
+ * linked to the Gallery editor UX: if they are changed here, they also
+ * must be changed in the HTML.
+ */
+var Effects = {
```
Comment (KevinGrandon) [inline apps/gallery/js/ImageEditor.js]: Moving this to make sure it's defined before use in functions. (All large blocks of changes in this PR are for the same reason)

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e7e364523ea5ed2d6d7bde9d1ac71f468856b96a

File: apps/gallery/js/ImageEditor.js [hunk 2]
```diff
@@ -229,134 +405,20 @@ function editAspectMoreHandler() {
...
-  }
-
-  return {
-    resize: resize,
-    forceSetExposure: forceSetExposure,
-    setExposure: setExposure,
-    getExposure: function() { return currentExposure; }
-  };
-})();
-
```
Comment (punamdahiya) [inline apps/gallery/js/ImageEditor.js]: nit: If JSHint allows below slider on change handler line 408 - 436 should be moved above with the exposureSlider code at Line 221

File: apps/gallery/js/gallery.js [hunk 3]
```diff
@@ -646,10 +704,12 @@ function share(blobs, blobName) {
   // If all the blobs are image we use 'image/*'. If all are videos
   // we use 'video/*'. Otherwise, 'application/*'.
   var type;
-  if (types.length === 1 || types.every(function(t) { return t === types[0]; }))
+  if (types.length === 1 ||
+    types.every(function(t) { return t === types[0]; })) {
```
Comment (punamdahiya) [inline apps/gallery/js/gallery.js]: nit: indent two spaces to align with top expression types.length === 1 for readibility

File: apps/gallery/js/ImageEditor.js [hunk 4]
```diff
@@ -1517,20 +1583,24 @@ ImageEditor.prototype.cropStart = function(ev) {
...
       // change), only allow the crop region to get bigger
       var newWidth = newright - newleft;
-      if ((newWidth < (region.right - region.left)) && (newWidth < minWidth))
+      if ((newWidth < (region.right - region.left)) && (newWidth < minWidth)) {
         return;
+      }
       var newHeight = newbottom - newtop;
-      if ((newHeight < (region.bottom - region.top)) && (newHeight < minHeight))
+      if ((newHeight < (region.bottom - region.top)) &&
+        (newHeight < minHeight)) {
```
Comment (punamdahiya) [inline apps/gallery/js/ImageEditor.js]: nit: indent two spaces  to align with the top expression (newHeight < (region.bottom - region.top)) for readability

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a0777610556da32d3fb0cda270a0aa8eaf59f0f2

File: apps/gallery/js/ImageEditor.js (hunk 2 above)
Comment (KevinGrandon) [inline apps/gallery/js/ImageEditor.js]: This is done, thanks!

**Pull request merged**